### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by goldenuni**

### CODE REVIEW

File: app/create_file.py [hunk 1]
```diff
@@ -1 +1,43 @@
...
+def create_file(name_file: str) -> None:
+    if os.path.exists(name_file):
+        flag = "a"
+    else:
+        flag = "w"
+    with open(name_file, flag) as file:
+        if flag == "w":
+            file.write(datetime.now().strftime("%Y-%m-%d %H:%M:%S"))
+        else:
+            file.write("\n" + datetime.now().strftime("%Y-%m-%d %H:%M:%S"))
```
Comment (ArsenPidhoretskyi) [inline app/create_file.py:16]:
```suggestion
        first_line = datetime.now().strftime("%Y-%m-%d %H:%M:%S")
        if flag != "w":
            first_line = "\n" + first_line
        file.write(first_line)
```

File: app/create_file.py [hunk 2]
```diff
@@ -1 +1,43 @@
...
+        line = ""
+        while line != "stop":
+            line = input("Enter content line: ")
+            if line != "stop":
+                file.write("\n" + line)
+                line = ""
```
Comment (ArsenPidhoretskyi) [inline app/create_file.py:22]: remove this

**Review by ArsenPidhoretskyi — CHANGES_REQUESTED** [review]

File: app/create_file.py [hunk 3]
```diff
@@ -1 +1,44 @@
...
+    if "-d" and "-f" in command:
+        if command.index("-d") < command.index("-f"):
+            start_index = command.index("-d")
+            finish_index = command.index("-f")
+            create_dirs(command[start_index + 1:finish_index])
+            create_file(command[-1])
+        else:
+            start_index = command.index("-d")
+            create_dirs(command[start_index + 1:])
+            create_file(command[command.index("-f") + 1])
```
Comment (ArsenPidhoretskyi) [inline app/create_file.py:39]:
```suggestion
       f_index = command.index("-f")
       d_index = command.index("-d")
        if d_index < f_index:
            create_dirs(command[d_index + 1:f_index])
            create_file(command[-1])
        else:
            create_dirs(command[d_index + 1:])
            create_file(command[f_index + 1])
```
No need so much times calculate f_index and d_index

**Review by ArsenPidhoretskyi — CHANGES_REQUESTED** [review]
Changes requested

File: app/create_file.py [hunk 4]
```diff
@@ -1 +1,46 @@
...
+
+
+if __name__ == "__main__":
+    command = sys.argv
+    if "-d" and "-f" in command:
+        d_index = command.index("-d")
+        f_index = command.index("-f")
+        if d_index < f_index:
+            start_index = d_index
+            finish_index = f_index
```
Comment (ArsenPidhoretskyi) [inline app/create_file.py:35]: Remove this variables....

File: app/create_file.py [hunk 5]
```diff
@@ -1 +1,46 @@
...
+            create_dirs(command[start_index + 1:finish_index])
+            create_file(command[-1])
+        else:
+            start_index = d_index
```
Comment (ArsenPidhoretskyi) [inline app/create_file.py:39]: Remove this line

**Review by ArsenPidhoretskyi — CHANGES_REQUESTED** [review]
Almost done. Just remove redundant code

**Review by Bohdan-Salamakha — CHANGES_REQUESTED** [review]
I should be able to create a file according to the task
https://github.com/mate-academy/py-create-file-from-terminal#:~:text=the%20file%20name.-,python%20create_file.py%20%2Df%20file.txt,-After%20pressing%20Enter
image

add row numeration like in example
https://github.com/mate-academy/py-create-file-from-terminal#:~:text=%2D02%2D01%2014%3A41%3A10-,1,-Line1%20content%0A2%20Line2%20content%0A3%20Line3%20content%0A%0A2022
image

File: app/create_file.py [hunk 6]
```diff
@@ -1 +1,45 @@
...
+
+def create_file(name_file: str) -> None:
+    flag = "a" if os.path.exists(name_file) else "w"
+    with open(name_file, flag) as file:
+        first_line = datetime.now().strftime("%Y-%m-%d %H:%M:%S")
+        if flag != "w":
+            first_line = "\n" + first_line
+        file.write(first_line)
+        line = ""
+        i = 1
```
Comment (ArsenPidhoretskyi) [inline app/create_file.py:15]: please rename i to line_number

File: app/create_file.py [hunk 7]
```diff
@@ -1 +1,45 @@
...
+        while line != "stop":
+            line = input("Enter content line: ")
+            if line != "stop":
+                file.write("\n" + str(i) + " " + line)
```
Comment (ArsenPidhoretskyi) [inline app/create_file.py:19]:
```suggestion
                file.write(f"\n{line_number}{line})
```

**Review by ArsenPidhoretskyi — CHANGES_REQUESTED** [review]
Well done!

File: app/create_file.py [hunk 8]
```diff
@@ -1 +1,45 @@
...
+        first_line = datetime.now().strftime("%Y-%m-%d %H:%M:%S")
+        if flag != "w":
+            first_line = "\n" + first_line
+        file.write(first_line)
+        line = ""
+        line_number = 1
+        while line != "stop":
+            line = input("Enter content line: ")
+            if line != "stop":
+                file.write("\n" + str(line_number) + " " + line)
```
Comment (ArsenPidhoretskyi) [inline app/create_file.py:19]: Please use f-string in the future. Concatenation is not good in this way

**Review by ArsenPidhoretskyi — APPROVED** [review]
Good job!

File: app/create_file.py (hunk 8 above)
Comment (goldenuni) [reply app/create_file.py:19]: Okey, I'll rember. Thanks)

**Pull request closed without merging**